### PR TITLE
[RATE-LIMIT] Unit Tests for Packet Callback Functions

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Brief Changelog
* Added unit tests for: 
  * ParseDenomFromRecvPacket
  * ParseDenomFromSendPacket
  * CheckRateLimit
  * SendRateLimitedPacket
  * ReceiveRateLimitedPacket
  * BeforeEpochStart 
* Changed all uints to sdk.Int

### CODE REVIEW

**Review by antstalepresh — CHANGES_REQUESTED** [review]
Great! just two comments here.

File: x/ratelimit/rate_limit_test.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+			Denom:     denom,
+			ChannelId: channelId,
+		},
+		Quota: &types.Quota{
+			MaxPercentSend: threshold,
+			MaxPercentRecv: threshold,
+		},
+		Flow: &types.Flow{
+			Inflow:       inflow,
+			Outflow:      outflow, // outflow almost at threshold
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/ratelimit/rate_limit_test.go:177]: done

File: x/ratelimit/keeper/rate_limit.go [hunk 2]
```diff
@@ -19,23 +19,36 @@ func (k Keeper) GetChannelValue(ctx sdk.Context, denom string) sdk.Int {
...
+		return rateLimit.Flow.AddInflow(amount, *rateLimit.Quota)
+	default:
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid packet direction (%s)", direction.String())
+	}
+}
+
 // Checks whether the given packet will exceed the rate limit
 // Called by OnRecvPacket and OnSendPacket
-func (k Keeper) CheckRateLimit(ctx sdk.Context, direction types.PacketDirection, denom string, channelId string, amount uint64) error {
+func (k Keeper) CheckRateLimit(ctx sdk.Context, direction types.PacketDirection, denom string, channelId string, amount sdk.Int) error {
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/ratelimit/keeper/rate_limit.go:36]: great callout! I'll adjust. What do you think about: `CheckTransferExceedsRateLimit`?

[the remark this answers is not in the document]
Comment (antstalepresh) [reply x/ratelimit/keeper/rate_limit.go:36]: Would be great if the function name involves meaning for updating something as the function does update rate limit object if transfer doesn't exceed the rate limit.

[the remark this answers is not in the document]
Comment (sampocs) [reply x/ratelimit/keeper/rate_limit.go:36]: pulling this offline

[the remark this answers is not in the document]
Comment (sampocs) [reply x/ratelimit/keeper/rate_limit.go:36]: chatted offline, agreed on `CheckRateLimitAndUpdateFlow`

**Review by antstalepresh — APPROVED** [review]

**Pull request merged**